aztec/src/main/kotlin/org/wordpress/aztec/source/InlineCssStyleFormatter.kt

```diff
@@ -36,21 +33,66 @@ class InlineCssStyleFormatter {
-            if (attributes.hasAttribute("style")) {
+            if (attributes.hasAttribute(STYLE_ATTRIBUTE)) {
 
                 if (start != end) {
-                    val style = attributes.getValue("", "style")
                     // Process the CSS 'color' property, remove any whitespace or newline characters
-                    val m = foregroundColorPattern.matcher(style.replace("\\s".toRegex(), ""))
-                    if (m.find()) {
-                        val colorString = m.group(1)
-                        val colorInt = ColorConverter.getColorInt(colorString)
-                        if (colorInt != ColorConverter.COLOR_NOT_FOUND) {
-                            text.setSpan(ForegroundColorSpan(colorInt), start, end, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
-                        }
-                    }
+                    val style = attributes.getValue("", STYLE_ATTRIBUTE).replace("\\s".toRegex(), "")
+
+                    processColor(style, text, start, end)
+                }
+            }
+        }
+
+        private fun getPattern(styleAttr: String): Pattern {
+            return Pattern.compile(
+                    "(?:;|\\A)$styleAttr:(.+?)(?:;|$)", Pattern.CASE_INSENSITIVE or Pattern.MULTILINE)
+        }
+
+        private fun processColor(style: String, text: Editable, start: Int, end: Int) {
+            val m = getPattern(CSS_COLOR_ATTRIBUTE).matcher(style)
+            if (m.find()) {
+                val colorString = m.group(1)
+                val colorInt = ColorConverter.getColorInt(colorString)
+                if (colorInt != ColorConverter.COLOR_NOT_FOUND) {
+                    text.setSpan(ForegroundColorSpan(colorInt), start, end, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
                 }
             }
         }
+
+        fun containsUnderlineDecorationStyle(attributes: AztecAttributes): Boolean {
+            if (attributes.hasAttribute(STYLE_ATTRIBUTE)) {
+                val style = attributes.getValue(STYLE_ATTRIBUTE).replace("\\s".toRegex(), "")
+                val m = getPattern(CSS_TEXT_DECORATION_ATTRIBUTE).matcher(style)
+                return m.find()
+            }
+            return false
+        }
+
+        fun removeStyle(attributes: AztecAttributes, styleAttributeName: String) {
+            if (attributes.hasAttribute(STYLE_ATTRIBUTE)) {
+                val style = attributes.getValue(STYLE_ATTRIBUTE).replace("\\s".toRegex(), "")
+                val m = getPattern(styleAttributeName).matcher(style)
+                val newStyle = m.replaceAll("")
+
+                if (newStyle.isBlank()) {
+                    attributes.removeAttribute(STYLE_ATTRIBUTE)
+                } else {
+                    attributes.setValue(STYLE_ATTRIBUTE, m.replaceAll(""))
+                }
+            }
+        }
+
+        fun addStyle(attributes: AztecAttributes, styleAttribute: String) {
```

Comment:
This is not used anywhere. Do you think we should keep it for future usage?

Reply:
This is sort of a utility class for CSS manipulation. I think it could be useful in the future if we support more styles.